firefox-ios/Client/Coordinators/Router/RouteBuilder.swift

```diff
@@ -34,6 +34,7 @@ final class RouteBuilder: FeatureFlaggable {
 
         if let host = parseURLHost(url) {
             let urlQuery = urlScanner.fullURLQueryItem()?.asURL
+            guard host.isValidURL(urlQuery: urlQuery) else { return nil }
```

Comment:
Does this check here covers the code below? Would be great to clean this up if possible so we have checks in one place only

```
           case .openUrl:
                // If we have a URL query, then make sure to check its a webpage
                if urlQuery == nil || urlQuery?.isWebPage() ?? false {
                    return .search(url: urlQuery, isPrivate: isPrivate)
                } else {
                    return nil
                }
```

Reply:
yes it does! cleaned it up, thanks for the catch!